### PR TITLE
Multiple-async-writes-to-stream fix, debug logging

### PR DESCRIPTION
**Pull request by timlaengert**
Fixed the error that was getting thrown by grpc, added a bit of debug logging.

### CODE REVIEW

**Review by daveismith — APPROVED** [review]

**Comment by deadwing88** [comment]
Async changes look good. 

For the logging we should have it as an env var that is passed into the container on startup instead of explicitly setting it in the Dockerfile. Would be nice to also start implementing different logging levels as needed (Information/Warning/Debug/Error) and stop using console print lines.

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-5.0

Happy to help with implementing this. We can also merge this and I can refactor after.

File: backend/ShootTheMoon/ShootTheMoon/Program.cs [hunk 1]
```diff
@@ -13,9 +13,14 @@ namespace ShootTheMoon
     public class Program
     {
         const int Port = 30051;
+        public static bool Verbose = false;
 
         public static void Main(string[] args)
         {
+            if (args.Length > 0)
```
Comment (deadwing88) [inline backend/ShootTheMoon/ShootTheMoon/Program.cs:20]: don't think you need this check before a foreach

**Comment by daveismith** [comment]
I'd say merge it for now and then go for a proper logging framework.

**Comment by timlaengert** [comment]
Ok yeah this all sounds good -- if you want to start bringing the logging into the 21st century, go for it :)

**Pull request merged**